src/tir/schedule/primitive/blockize_tensorize.cc

```diff
@@ -0,0 +1,638 @@
+                                                      const PrimExpr& predicate) {
+  if (!is_one(predicate)) return {};
+  std::vector<Array<arith::IterMark>> res;
+  std::unordered_set<const VarNode*> outer_loop_vars;
+  std::unordered_set<const VarNode*> inner_loop_vars;
+  for (const Var& var : outer_iters) {
+    outer_loop_vars.insert(var.get());
+  }
+  for (const Var& var : inner_iters) {
+    inner_loop_vars.insert(var.get());
+  }
+  const arith::IterMark unit_iter_mark(arith::IterSumExpr({}, 0), 1);
+
+  for (size_t i = 0; i < bindings.size(); ++i) {
+    bool outer = UsesVar(
+        bindings[i], [&outer_loop_vars](const VarNode* var) { return outer_loop_vars.count(var); });
+    bool inner = UsesVar(
+        bindings[i], [&inner_loop_vars](const VarNode* var) { return inner_loop_vars.count(var); });
+    arith::IterMark iter_mark;
+    if (bindings[i]->IsInstance<VarNode>()) {
+      iter_mark = arith::IterMark(
+          arith::IterSplitExpr(arith::IterMark(bindings[i], iter_vars[i]->dom->extent)),
+          iter_vars[i]->dom->extent);
+    } else {
+      iter_mark = arith::IterMark(arith::IterSumExpr({}, bindings[i]), iter_vars[i]->dom->extent);
+    }
+    if (outer && !inner) {
+      arith::IterMark outer{nullptr};
+      const auto& outer_iter = iter_mark;
+      const auto& inner_iter = unit_iter_mark;
+      res.push_back({outer_iter, inner_iter});
+    } else if (inner && !outer) {
+      const auto& outer_iter = unit_iter_mark;
+      const auto& inner_iter = iter_mark;
+      res.push_back({outer_iter, inner_iter});
+    } else if (!outer && !inner) {
+      const auto& outer_iter = unit_iter_mark;
+      const auto& inner_iter = unit_iter_mark;
+      res.push_back({outer_iter, inner_iter});
+    } else {
+      return {};
+    }
+  }
+  res.push_back({arith::IterMark(arith::IterSumExpr({}, 0), Bool(true)),
+                 arith::IterMark(arith::IterSumExpr({}, 0), Bool(true))});
+  return res;
+}
+
+/*!
+ * \brief Generate the blockized init block.
+ * \param block The original block with init.
+ * \param inner_block_realize The block realize of the inner block after blockize.
+ * \param inner_loops The inner loops after blockize.
+ * \return The subtree of the init block and its outer loops.
+ */
+Stmt GenerateBlockizedInit(const Block& block, const BlockRealize& inner_block_realize,
+                           const std::vector<const ForNode*>& inner_loops) {
+  Array<IterVar> init_block_iters;
+  Array<PrimExpr> init_bindings;
+  const Block& inner_block = inner_block_realize->block;
+
+  // Step 1: Collect data-parallel block iters
+  for (size_t i = 0; i < inner_block->iter_vars.size(); i++) {
+    const IterVar& iter_var = inner_block->iter_vars[i];
+    const PrimExpr& binding = inner_block_realize->iter_values[i];
+    if (iter_var->iter_type == IterVarType::kDataPar &&
+        UsesVar(block->init.value(),
+                [&iter_var](const VarNode* var) { return var == iter_var->var.get(); })) {
```

Comment:
nit: to make the logic look more natural

```suggestion
                [tgt_var = iter_var->var.get()](const VarNode* var) { return var == tgt_var; })) {
```